marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/savedQueries/saveSavedQueryTest.sjs

```diff
@@ -0,0 +1,82 @@
+const test = require("/test/test-helper.xqy");
+
+var saveQuery = JSON.stringify({
+  "savedQuery": {
+    "id": "",
+    "name": "some-query",
+    "description": "some-query-description",
+    "query": {
+      "searchText": "some-string",
+      "entityTypeIds": [
+        "Entity1"
+      ],
+      "selectedFacets": {
+        "Collection": {
+          "dataType": "string",
+          "stringValues": [
+            "Entity1",
+            "Collection1"
+          ]
+        },
+        "facet1": {
+          "dataType": "decimal",
+          "rangeValues": {
+            "lowerBound": "2.5",
+            "upperBound": "15"
+          }
+        },
+        "facet2": {
+          "dataType": "dateTime",
+          "rangeValues": {
+            "lowerBound": "2020-01-01T13:06:17",
+            "upperBound": "2020-01-22T13:06:17"
+          }
+        }
+      }
+    },
+    "propertiesToDisplay": ["facet1", "EntityTypeProperty1"]
+  }
+});
+
+function invokeService(saveQuery) {
+  return fn.head(xdmp.invoke(
+      "/data-hub/5/data-services/savedQueries/saveSavedQuery.sjs",
+      {
+        "saveQuery": saveQuery
+      }
+  ));
+}
+
+
+function testSaveNewQuery() {
+  const result = invokeService(saveQuery);
```

Comment:
This test should also confirm the expected URI, permissions, and collections on the saved query.

Reply:
the complete js file is run as a single transaction. So cannot get the document until the transaction is committed. Added the tests in the java layer